### PR TITLE
[1356] Progress commit for review.

### PR DESCRIPTION
**Pull request by mark-terry**

### CODE REVIEW

File: data/beaconrestapi/build.gradle [hunk 1]
```diff
@@ -2,6 +2,7 @@ dependencies {
     implementation project(':data')
     implementation project(':data:provider')
     implementation project(':ethereum:datastructures')
+    implementation project(':ethereum:statetransition')
```
Comment (rolfyone) [inline data/beaconrestapi/build.gradle:5]: can we avoid this? i'd really prefer the api isn't pulling in something as low as statetransition - anything at that level should be in providers.

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/validator/PostBlock.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            jsonProvider.objectToJSON(blockImportResult.getFailureReason().get().getMessage()));
+      } else {
+        ctx.result(jsonProvider.objectToJSON(blockImportResult.getHash_tree_root()));
+      }
+    } catch (final JsonMappingException ex) {
+      ctx.status(SC_BAD_REQUEST);
+      ctx.result(jsonProvider.objectToJSON(new BadRequest(ex.getMessage())));
+    } catch (final Exception ex) {
+      ctx.status(SC_INTERNAL_SERVER_ERROR);
+    }
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/validator/PostBlock.java:97]: that should be handled by the catchall, we shouldn't need to do that here...

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/handlers/validator/PostBlockTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+package tech.pegasys.artemis.beaconrestapi.handlers.validator;
+
+import org.junit.jupiter.api.Test;
+
+class PostBlockTest {
+  @Test
+  public void shouldConvertSchemaToInternal() {
+
+  }
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/handlers/validator/PostBlockTest.java:22]: awfully empty...

**Comment by mark-terry** [comment]
Outdated and not required.

**Pull request closed without merging**